extensions/knative/generator/cmd/generate/generate.go

```diff
@@ -16,169 +16,91 @@
-		{"knative.dev/pkg/tracker", "tracker", "io.fabric8.knative.tracker", "knative_tracker_"},
-		{"github.com/knative/pkg/apis", "v1", "io.fabric8.knative.v1", "knative_"},
-		{"knative.dev/eventing/pkg/apis/duck/v1beta1", "duck", "io.fabric8.knative.dev.duck.v1beta1", "knative_dev_duck_v1beta1_"},
-		{"knative.dev/eventing/pkg/apis/duck/v1alpha1", "duck", "io.fabric8.knative.dev.duck.v1alpha1", "knative_dev_duck_v1alpha1_"},
-		{"knative.dev/pkg/apis/duck/v1", "duck", "io.fabric8.knative.duck.v1", "knative_dev_duck_v1_"},
-    {"knative.dev/pkg/apis/duck/v1beta1", "duck", "io.fabric8.knative.duck.v1beta1", "knative_dev_duck_v1beta1_"},
-    {"knative.dev/pkg/apis/duck/v1alpha1", "duck", "io.fabric8.knative.duck.v1alpha1", "knative_dev_duck_v1alpha1_"},
-		{"github.com/knative/pkg/apis/duck/v1", "duck", "io.fabric8.knative.duck.v1", "knative_duck_v1_"},
-		{"github.com/knative/serving/pkg/apis/serving/v1alpha1", "serving", "io.fabric8.knative.serving.v1alpha1", "knative_serving_v1alpha1_"},
-		{"github.com/knative/serving/pkg/apis/serving/v1beta1", "serving", "io.fabric8.knative.serving.v1beta1", "knative_serving_v1beta1_"},
-		{"github.com/knative/serving/pkg/apis/serving/v1", "serving", "io.fabric8.knative.serving.v1", "knative_serving_v1_"},
-		{"knative.dev/serving/pkg/apis/serving/v1", "serving", "io.fabric8.knative.serving.v1", "knative_serving_v1_"},
-		{"knative.dev/eventing/pkg/apis/messaging/v1beta1", "messaging", "io.fabric8.knative.dev.messaging.v1beta1", "knative_dev_messaging_v1beta1_"},
-		{"knative.dev/eventing/pkg/apis/duck/v1beta1", "duck", "io.fabric8.knative.dev.duck.v1beta1", "knative_dev_duck_v1beta1_"},
-		{"github.com/knative/eventing/pkg/apis/eventing/v1alpha1", "eventing", "io.fabric8.knative.eventing.v1alpha1", "knative_eventing_v1alpha1_"},
-		{"github.com/knative/eventing/pkg/apis/messaging/v1alpha1", "messaging", "io.fabric8.knative.messaging.v1alpha1", "knative_messaging_v1alpha1_"},
-    {"github.com/knative/eventing/pkg/apis/messaging/v1beta1", "messaging", "io.fabric8.knative.messaging.v1beta1", "knative_messaging_v1beta1_"},
-		{"github.com/knative/eventing/pkg/apis/duck/v1alpha1", "duck", "io.fabric8.knative.duck.v1alpha1", "knative_duck_v1alpha1_"},
-    {"github.com/knative/eventing/pkg/apis/duck/v1beta1", "duck", "io.fabric8.knative.duck.v1beta1", "knative_duck_v1beta1_"},
-    {"github.com/knative/eventing/pkg/apis/flows/v1alpha1", "flows", "io.fabric8.knative.flows.v1alpha1", "knative_flows_v1alpha1_"},
-    {"github.com/knative/eventing/pkg/apis/legacysources/v1alpha1", "legacysources", "io.fabric8.knative.legacysources.v1alpha1", "knative_legacysources_v1alpha1_"},
-	}
 
-	typeMap := map[reflect.Type]reflect.Type{
-		reflect.TypeOf(time.Time{}): reflect.TypeOf(""),
-		reflect.TypeOf(struct{}{}):  reflect.TypeOf(""),
+	// the CRD List types for which the model should be generated
+	// no other types need to be defined as they are auto discovered
+	crdLists := []reflect.Type{
+
+		// serving v1
```

Comment:
Umm, why have you removed all resources from `duck v1beta1`. Are they not part of Knative model?

Reply:
2 things:
- all types that are only used internally are auto-discovered by the generator. Only CRDs need to be passed in as a starting point.
- AFAIK everything related to duck is not a CRD exposed by knative. Instead that is somehow used internally for duck typing. I might be wrong on this, do you have other information?

Reply:
Umm, I don't have much idea about Knative. I think it's okay if duck classes were not exposed earlier.